sirepo/package_data/static/json/raydata-schema.json

```diff
@@ -14,48 +14,44 @@
             ]
         }
     },
-    "enum": {
-        "CatalogName": [
-            ["chx", "chx"],
-            ["csx", "csx"]
-        ]
-    },
+    "enum": {},
     "frameIdFields": {
         "animation": []
     },
     "localRoutes": {
-        "analysis": {
-            "route": "/analysis/:simulationId",
+        "analysisQueue": {
+            "route": "/analysis-queue/:simulationId",
             "config": {
-                "controller": "AnalysisController as analysis",
-                "templateUrl": "/static/html/raydata-analysis.html"
+                "templateUrl": "/static/html/raydata-analysis-queue.html"
           }
         },
-        "dataSource": {
-            "route": "/data-source/:simulationId",
+        "analysisCompleted": {
+            "route": "/analysis-completed/:simulationId",
             "config": {
                 "controller": "DataSourceController as dataSource",
```

Comment:
I get an error when I try to remove it, and I don't understand why it's causing this.

```
../sirepo/server.py:159:api_errorLogging 10.10.10.1: javascript error: {
    "cause": "<no cause>",
    "message": "TypeError: Cannot read properties of undefined (reading 'simulationId')",
    "stackTrace": "TypeError: Cannot read properties of undefined (reading 'simulationId')
    at sendWithSimulationFields (http://v.radia.run:8000/static/js/sirepo.js?20220912.205016:2100:77)
    at self.sendStatelessCompute (http://v.radia.run:8000/static/js/sirepo.js?20220912.205016:2464:9)
    at controller (http://v.radia.run:8000/static/js/raydata.js?20220912.205016:168:27)
    at Object.invoke (http://v.radia.run:8000/static/js/ext/angular.min.js?20220912.205016:45:62)
    at S.instance (http://v.radia.run:8000/static/js/ext/angular.min.js?20220912.205016:97:486)
    at p (http://v.radia.run:8000/static/js/ext/angular.min.js?20220912.205016:71:457)
    at g (http://v.radia.run:8000/static/js/ext/angular.min.js?20220912.205016:64:477)
    at p (http://v.radia.run:8000/static/js/ext/angular.min.js?20220912.205016:73:75)
    at g (http://v.radia.run:8000/static/js/ext/angular.min.js?20220912.205016:64:477)
    at http://v.radia.run:8000/static/js/ext/angular.min.js?20220912.205016:64:100",
    "type": "clientException",
    "url": "http://v.radia.run:8000/raydata#/analysis-completed/bPgE72es"
}
```

Reply:
Ah here is what's going on:
- We aren't supplying simulationId in data so it is coming from appState (https://github.com/radiasoft/sirepo/blob/9f260b30ffd91aa6676eddbc1a32e37acdc6f1f8/sirepo/package_data/static/js/sirepo.js#L2100)
- If you srdbg appState.models you'll see that it is an empty object. That is where the `Cannot read properties of undefined` error arises
- If models are empty that probably means they haven't been loaded from the server yet. 
- I searched in sirepo.js for "localRoutes" because we are removing the controller field from a localRoute object. That lead to this code https://github.com/radiasoft/sirepo/blob/9f260b30ffd91aa6676eddbc1a32e37acdc6f1f8/sirepo/package_data/static/js/sirepo.js#L205 . We only add a "simulationDetailTemplate" if we have a controller (what we are removing).
- Inside that simulationDetailTemplate there is an ng-if based on appState.isLoaded() https://github.com/radiasoft/sirepo/blob/9f260b30ffd91aa6676eddbc1a32e37acdc6f1f8/sirepo/package_data/static/js/sirepo-components.js#L28 .

So when we include a controller then we get an ng-if on appState being loaded. No code after is executed until the appState is loaded. Since we removed the controller we removed that protection and we are trying to get simulationId off of appState before it is loaded.

So I think we have a few options:
1. keep the controller so we keep this behavior
2. Protect all of our calls with `appState.isLoaded/whenModelsLoaded`
3. Add some other structure so we don't have to have a controller to get the appState.isLoaded protection

If we weren't building the React ui my vote would be for 3. Nothing in Sirepo can really happen until appState.isLoaded so that should be included for free from the top-level. But, since we are building a new ui and hopefully fixing this type of behavior let's do 1 and just have an empty controller. Sound good?

Reply:
The bug on the Queue tab you mentioned in your comment above seems to be caused by a similar issue. I added controllers for Queue and Replay tabs and the bug went away. Is that an alright solution?